src/common/cluster/cluster.ts

```diff
@@ -217,9 +217,16 @@ export class Cluster implements ClusterModel, ClusterState {
     return this.preferences.defaultNamespace;
   }
 
-  constructor(private readonly dependencies: ClusterDependencies, model: ClusterModel) {
+  constructor(private readonly dependencies: ClusterDependencies, { id, ...model }: ClusterModel) {
     makeObservable(this);
-    this.id = model.id;
+
+    const { error } = clusterModelIdChecker.validate({ id });
+
+    if (error) {
+      throw error;
```

Comment:
I assume there's something ready to catch this?

Reply:
Yes because there are already other errors that can be thrown in this constructor. For instance on line 235 below.